compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -469,14 +469,22 @@ pub trait PrintState<'a>: std::ops::Deref<Target = pp::Printer> + std::ops::Dere
                 true,
                 span,
             ),
-            MacArgs::Empty | MacArgs::Eq(..) => {
+            MacArgs::Empty => {
                 self.print_path(&item.path, false, 0);
-                if let MacArgs::Eq(_, token) = &item.args {
-                    self.space();
-                    self.word_space("=");
-                    let token_str = self.token_to_string_ext(token, true);
-                    self.word(token_str);
-                }
+            }
+            MacArgs::Eq(_, MacArgsEqKind::Ast(expr)) => {
+                self.print_path(&item.path, false, 0);
+                self.space();
+                self.word_space("=");
+                let token_str = self.expr_to_string(expr);
+                self.word(token_str);
```

Comment:
```suggestion
                self.print_expr(expr);
```

Reply:
It doesn't work:
```
error[E0599]: no method named `print_expr` found for mutable reference `&mut Self` in the current scope
   --> compiler/rustc_ast_pretty/src/pprust/state.rs:479:22
    |
479 |                 self.print_expr(expr);
    |                      ^^^^^^^^^^ method not found in `&mut Self`
```
This code is within a provided method within trait `PrintState`. `print_expr` is an inherent method in `State`.